Dockerfile

```diff
@@ -12,6 +12,15 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
 RUN python -m venv /venv
 ENV PATH=/venv/bin:$PATH
 
+ENV OTLP_EXPORT_ENABLED=false
+# enable opentelemetry support
+ENV OTEL_EXPORTER_OTLP_TRACES_PROTOCOL=http/protobuf
+# Change this to point to Jaeger server before merging e.g. https://daq-services-jaeger
+ENV OTEL_EXPORTER_OTLP_ENDPOINT=http://127.0.0.1:4318
+# Ensure that all Http headers are captured
+ENV OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_REQUEST=".*"
+ENV OTEL_INSTRUMENTATION_HTTP_CAPTURE_HEADERS_SERVER_RESPONSE=".*"
```

Comment:
How does that work for the dev container though as it uses the dockerfile to initialise? The config is set in the helm chart too but that doesn't help in VSCode?

Reply:
For the devcontainer you should be editing `.devcontainer/devcontainer.json`, which configures the base image into a devcontainer, I think you need to add something that looks a bit like this:

```json
    "containerEnv": {
        "OTLP_EXPORT_ENABLED": false,
        etc
        etc
    },
```

Reply:
As noted above, not setting the variable at all will result in Export being off, which is the desired default state, however the other vars are needed to configure tracing correctly, so that if export is enabled it can function. I guess therefore these should be moved into the devcontainer.json as you say, but where can I set them for a plain valnilla blueapi being run from the command line with no Dev Container or hekm chart involved?

Reply:
I think at that point they should be passed in at container startup time

```
docker run -d --rm -e OLTP_FOO=true -e OLTP_BAR=false blueapi:latest
```

Reply:
We should, in general, update the docs to reflect this PR. I don't mind if you do that here or make a separate issue

Reply:
https://github.com/DiamondLightSource/blueapi/issues/682 created to cover this